### DISCUSSION TITLE
Can the html slider plugin be altered to show the value?

### DISCUSSION THREAD

**Discussion by SaoirseConnorDesai** [jspsych/jsPsych#963] [Q&A] [answered]
I would like participants to be able to see which value the slider is on to get a more fine-grained measure. Is there a way to adapt the slider plugin to do this? I found something on w3Schools but I'm unsure of how to adapt the plugin to have this feature. 

Many thanks
Saoirse

**Comment by vijaymarupudi** [comment, marked as answer]
I think there's a way without editing the plugin. In the html input to the plugin, you can include an element to display the value. Something like this?

```html
<p>Your answer is <span id="answer-display"></span></p>
```

And when you're adding the plugin to the timeline, add an [on_load] callback with this:

[on_load]: https://www.jspsych.org/overview/callbacks/#on_load

```javascript
timeline.push({
  type: 'html-slider-response',
  html: '', // your stimulus
  on_load: function()  {
    var input = document.querySelector('#jspsych-html-slider-response-response')

    input.addEventListener('input', function() {
      document.querySelector('#answer-display').innerText = input.valueAsNumber.toString()
    })

  }
})
```

**Comment by SaoirseConnorDesai** [comment]
Thank you!
